### PR TITLE
fix(trns): reset running qty to BALANCE in position trade totals

### PR DESCRIPTION
**Pull request by monowai**
## Problem
Position drill-down → trades (`/trns/trades/[portfolioId]/[assetId]`) summed every `trn.quantity` for the footer "Total Qty". A `BALANCE` transaction states the **absolute** position as at its trade date (no cash impact), so blindly adding it double-counts the running total.

## Fix
- New `computeTradeGroupTotals(transactions)` in `tradeUtils.ts`:
  - Sorts chronologically (trades endpoint returns `tradeDate DESC`), so it is order-independent.
  - On `BALANCE`, running quantity **resets** to the balance qty; subsequent trades accumulate on top. Latest BALANCE wins.
  - Cash-side totals (tradeAmount/cashAmount/fees/tax) remain plain sums (BALANCE has no cash impact).
- `[...trades].tsx` `groupedByBroker` now folds via the helper instead of inline `+=`.

## Tests
- Red→Green TDD. 6 new cases: no-balance sum, reset, post-balance accumulation, order-independence, multi-balance latest-wins, cash totals still sum.
- 243/243 trns suite, typecheck, eslint, prettier clean.

## Not covered
Not exercised in a live browser (needs a demo position carrying a BALANCE trn). Logic is pure and unit-tested against the real DESC ordering.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Tests**
  * Added comprehensive test coverage for trade group total calculations, including edge cases around `BALANCE` handling and order independence.
* **Refactor**
  * Updated trade aggregation on the Trades page to use a shared totals helper instead of inline accumulation, reducing duplication.
* **Bug Fixes**
  * Corrected per-broker totals to apply `BALANCE` quantity/reset and latest-`BALANCE` precedence consistently, regardless of transaction order.

### CODE REVIEW

File: src/lib/utils/trns/tradeUtils.ts [hunk 1]
```diff
@@ -416,3 +416,40 @@ export const convert = (tradeFormData: TradeFormData): string => {
...
+  transactions: Transaction[],
+): TradeGroupTotals =>
+  [...transactions]
+    .sort((a, b) => a.tradeDate.localeCompare(b.tradeDate))
+    .reduce<TradeGroupTotals>(
+      (totals, trn) => ({
+        quantity:
+          trn.trnType === "BALANCE"
+            ? trn.quantity || 0
+            : totals.quantity + (trn.quantity || 0),
```
[the remark this answers is not in the document]
Comment (monowai) [reply src/lib/utils/trns/tradeUtils.ts]: Done in 1536bd2. Same-date tiebreak now deterministic: a BALANCE is the authoritative as-at-date position, so non-BALANCE trades sharing its date are subsumed — BALANCE sorts last on equal dates so the reset wins. Added a test asserting both input orders yield the balance qty.

**Comment by monowai** [comment]
Re: import alias suggestion (`types/beancounter` → `@types/beancounter`) — declining. The `@types/*` alias hits TS6137 for `.d.ts` resolution in this repo; the bare `types/beancounter` import resolves via baseUrl and matches the existing convention (the pre-existing `TradeFormData` import in the same file is already bare). Keeping it as-is.

**Comment by monowai** [comment]
Nitpicks:
- `@types/beancounter` alias in both `tradeUtils.ts` and `tradeUtils.test.ts` imports — declining (same TS6137 `.d.ts` resolution issue; bare `types/beancounter` is repo convention and matches the pre-existing `TradeFormData` import).
- Same-date BALANCE / regular-trade ordering test — already added in 1536bd2 ("BALANCE wins over a same-date trade regardless of input order", asserts both input orders).
